src/http-agent.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { AxiosRequestConfig } from 'axios';
+import { HttpProxyAgent, HttpsProxyAgent } from 'hpagent';
+
+export function getHttpAgents(
+  proxyUrl?: URL,
```

Comment:
in the PoC we also handled CA (Certificate authority) here, do we need a subtask to handle this?

Reply:
I think it fits in [a separate task](https://sonarsource.atlassian.net/browse/SCANNPM-17), as we won't be able to redo it as in the POC and will have to rely on the new bootstrapper properties. I was thinking of taking the task afterward